src/miniflask/miniflask.py

```diff
@@ -632,17 +632,23 @@ def parse_args(self,  # noqa: C901 too-complex  pylint: disable=too-many-stateme
             argv.remove("--help")
             print_help = True
 
-        # split --varname=value expressions
-        argv = [v for val in argv for v in (val.split("=", 1) if val.startswith("--") or val.startswith("-") and not val[1:].isnumeric() else [val])]
+        # split `--varname=value` expressions to `--varname value`
```

Comment:
Can we add a comment on 'why' this is actually needed?

Reply:
AFAIK, argparse does allow key-value pairs using the syntax `-variable=value`, but not for double-dashes, `--variable=value`. (?)

So either, we have to rewrite all double-dash calls to single-dash calls at this point or split each argument at the first occurrence of an equal sign.

I have added a note at this specific position.